### PR TITLE
Document optional declaration of  io.quarkus.qute.i18n.Message.value()

### PR DESCRIPTION
**Pull request by michalvavrik**
Documents: #17857

### CODE REVIEW

**Comment by michalvavrik** [comment]
cc @mkouba

File: docs/src/main/asciidoc/qute-reference.adoc [hunk 1]
```diff
@@ -2196,6 +2196,43 @@ public class MyBean {
 ----
 <1> The annotation value is a locale tag string (IETF).
 
+==== Message Value and Localized File
```
Comment (mkouba) [inline docs/src/main/asciidoc/qute-reference.adoc:2199]: I'd use the title _"Message Templates"_...

File: docs/src/main/asciidoc/qute-reference.adoc [hunk 2]
```diff
@@ -2196,6 +2196,43 @@ public class MyBean {
...
+
+Every annotated method of the *message bundle interface* must define a value. The value is normally defined by `io.quarkus.qute.i18n.Message#value()`,
```
Comment (mkouba) [inline docs/src/main/asciidoc/qute-reference.adoc:2201]: _"Every method of a message bundle interface must define a message template."_

File: docs/src/main/asciidoc/qute-reference.adoc [hunk 3]
```diff
@@ -2196,6 +2196,43 @@ public class MyBean {
...
+[source,properties]
+----
+hello_name=Hello \
+   {name} and \
+   good morning!
+goodbye=Best regards, {name} <1>
+----
+<1> The value is ignored as `io.quarkus.qute.i18n.Message#value()` is always prioritized.
+
+Message values are validated during the build time, if missing value for any message key is detected, an exception
```
Comment (mkouba) [inline docs/src/main/asciidoc/qute-reference.adoc:2233]: _"Message templates are validated during the build. If a missing message template is detected, ..."_

**Review by mkouba — COMMENTED** [review]
Looks good. I've added some minor comments...

File: docs/src/main/asciidoc/qute-reference.adoc (hunk 1 above)
Comment (michalvavrik) [reply docs/src/main/asciidoc/qute-reference.adoc]: I hope I got ellipsis right. Thanks

**Comment by michalvavrik** [comment]
Fixed, thank you for feedback.

**Review by mkouba — APPROVED** [review]

**Comment by mkouba** [comment]
Merged, thanks!

**Pull request merged**